src/event/writer.rs

```diff
@@ -118,10 +123,22 @@ impl EventWriter {
     pub async fn write_event(&mut self, event: Vec<u8>) -> oneshot::Receiver<Result<(), Error>> {
         let size = event.len();
         let (tx, rx) = oneshot::channel();
+        let (tx_flush, rx_flush) = oneshot::channel();
         let routing_info = RoutingInfo::RoutingKey(None);
+        let routing_info_flush = RoutingInfo::RoutingKey(None);
         if let Some(pending_event) = PendingEvent::with_header(routing_info, event, None, tx) {
             let append_event = Incoming::AppendEvent(pending_event);
-            self.writer_event_internal(append_event, size, rx).await
+            let write_event = self.writer_event_internal(append_event, size, rx).await;
+            if let Some(pending_event_flush) =
+                PendingEvent::without_header(routing_info_flush, Vec::new(), None, tx_flush)
+            {
+                let append_event_flush = Incoming::AppendEvent(pending_event_flush);
+                let flush_rec = self.writer_event_internal(append_event_flush, 0, rx_flush).await;
+                self.event_handles.push_back(flush_rec);
+            } else {
+                panic!("Failed to track event in flush.");
```

Comment:
@tkaitchuck is there a better way to solve this ?

Reply:
`.expect()`